### ISSUE TITLE
Bug: valid-title can't eliminate string types with settings.typecheck

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [vitest-dev/eslint-plugin-vitest#681]
`settings.typecheck` allows `vitest/valid-title` to skip reporting on class and function types, as determined by `services.getTypeAtLocation`:

https://github.com/vitest-dev/eslint-plugin-vitest/blob/c808a210344b0a10eea8442fb4711be874d9f0f4/src/rules/valid-title.ts#L211-L217

But if the discovered type is a string then it may still get reported on.

```ts
import { describe, test, expect } from "vitest";

declare const outerName: string;

describe(outerName, () => {
  //     ~~~~~~~~~
  // Test title must be a string, a function or class name
  test("item", () => {
    expect(0).toBe(0);
  });
});
```
